grpcbp/client_middlewares_test.go

```diff
@@ -0,0 +1,174 @@
+package grpcbp
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"testing"
+	"time"
+
+	pb "github.com/grpc-ecosystem/go-grpc-middleware/testing/testproto"
+	"github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/mqsend"
+	"github.com/reddit/baseplate.go/tracing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestMonitorInterceptor(t *testing.T) {
+	// create test server
+	l, _ := initServer(t)
+
+	// create test client with monitoring interceptor
+	conn := initClient(t, l, grpc.WithUnaryInterceptor(
+		MonitorInterceptor(
+			MonitorClientArgs{
+				ServiceSlug: "test",
+			},
+		),
+	))
+
+	// instantiate gRPC client
+	ctx, mmq := initServerSpan(t)
+	client := pb.NewTestServiceClient(conn)
+
+	t.Run("span-success", func(t *testing.T) {
+		_, err := client.Ping(ctx, &pb.PingRequest{})
+		if err != nil {
```

Comment:
`if _, err := ...; err != nil {` to make it clear that you don't need the err later